src/collector.ts

```diff
@@ -58,51 +51,61 @@ export async function runCollector(
       };
       sendEvent(worker, event);
     }
-    if (ledgersProcessed % 60 === 0) {
-      // approx every 5m
-      // send an oracle scan event
-      const event: OracleScanEvent = {
-        type: EventType.ORACLE_SCAN,
-        timestamp: Date.now(),
-      };
-      sendEvent(worker, event);
-    }
-    if (ledgersProcessed % 1203 === 0) {
-      // approx every 2hr
-      // send a user update event to update any users that have not been updated in ~2 weeks
-      const event: UserRefreshEvent = {
-        type: EventType.USER_REFRESH,
-        timestamp: Date.now(),
-        cutoff: Math.max(latestLedger - 14 * 17280, 0),
-      };
-      sendEvent(worker, event);
-    }
-    if (ledgersProcessed % 1207 === 0) {
-      // approx every 2hr
-      // send a liq scan event
-      const event: LiqScanEvent = {
-        type: EventType.LIQ_SCAN,
+
+    for (const poolConfig of poolConfigs) {
+      // new ledger detected
+      const ledger_event: LedgerEvent = {
+        type: EventType.LEDGER,
         timestamp: Date.now(),
+        ledger: latestLedger,
+        poolConfig,
       };
-      sendEvent(worker, event);
+      sendEvent(bidder, ledger_event);
+      // send long running work events to worker
+      if (ledgersProcessed % 60 === 0) {
+        // approx every 5m
+        // send an oracle scan event
+        const event: OracleScanEvent = {
+          type: EventType.ORACLE_SCAN,
+          timestamp: Date.now(),
+          poolConfig,
+        };
+        sendEvent(worker, event);
+      }
+      if (ledgersProcessed % 1203 === 0) {
+        // approx every 2hr
+        // send a user update event to update any users that have not been updated in ~2 weeks
+        const event: UserRefreshEvent = {
+          type: EventType.USER_REFRESH,
+          timestamp: Date.now(),
+          cutoff: Math.max(latestLedger - 14 * 17280, 0),
+          poolConfig,
+        };
```

Comment:
does this event really need a pool? It just loads old users, and attempts to update them.

IIRC the logic should skip failed updates anyway, and continue on down the rest of the list, so there is no risk of a borked pool oracle or something causing a massive holdup.

This being said, it might make sense to send an alert if the oldest entry is 
older than (cutoff - 6hr old), as that means we have a stale user that is not getting properly updated for some reason.

Reply:
There could be some issues with doing this for a large number of pools at once too, given the events will process "concurrently"

Reply:
The soroban helper will cache duplicate pool reads too, so that is not an issue. For oracle prices, it might be nice to cache a failure to load too, that way we don't spam the RPC trying to load this for all old users, and this could be v important for `liqScan`, if there is a big market downturn?